### PR TITLE
NAS-115184 / 22.12 / NAS-115184: Refactoring Groups

### PR DESCRIPTION
**Pull request by denysbutenko**
For testing go to `Credentials / Local Groups`, try to add/edit/remove a group, update members, etc

### CODE REVIEW

**Review by RehanY147 — CHANGES_REQUESTED** [review]

File: src/app/pages/account/groups/group-list/group-list.component.ts [hunk 1]
```diff
@@ -1,216 +1,147 @@
...
+    title: this.translate.instant('Loading...'),
+  };
+  errorConfig: EmptyConfig = {
+    type: EmptyType.Errors,
+    large: true,
+    title: this.translate.instant('Can not retrieve response'),
+  };
+  expandedRow: Group;
+  @ViewChildren(IxDetailRowDirective) private detailRows: QueryList<IxDetailRowDirective>;
+  error$ = this.store$.select(selectGroupState).pipe(map((state) => state.error));
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/pages/account/groups/group-list/group-list.component.ts:59]: Yeah, good point!

File: src/app/pages/account/groups/groups.routing.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    children: [
+      {
+        path: '',
+        component: GroupListComponent,
+        data: { title: 'Groups', breadcrumb: 'Groups' },
+      },
+      {
+        path: ':pk/members',
+        component: GroupMembersComponent,
+        data: { title: 'Update Members', breadcrumb: 'Members' },
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/pages/account/groups/groups.routing.ts:19]: Probably some issues in the breadcrumb generation. Checking details

**Review by RehanY147 — APPROVED** [review]
Looks good. Changes are persistent and form submits successfully. One minor note, the "Save" button doesn't disable while the form is submitting. It's possible to keep clicking the button multiple times.

**Comment by denysbutenko** [comment]
> Looks good. Changes are persistent and form submits successfully. One minor note, the "Save" button doesn't disable while the form is submitting. It's possible to keep clicking the button multiple times.

Good note, thanks!

**Pull request merged**